awx/playbooks/project_update.yml

```diff
@@ -139,16 +145,18 @@
       delegate_to: localhost
 
     - block:
-      - name: detect collections/requirements.yml
-        stat: path={{project_path|quote}}/collections/requirements.yml
+      - name: detect collections/requirements.(yml/yaml)
+        stat: path={{project_path|quote}}/collections/requirements{{ item.ext }}
+        with_items: "{{ yaml_exts }}"
         register: doesCollectionRequirementsExist
 
-      - name: fetch galaxy collections from collections/requirements.yml
-        command: ansible-galaxy collection install -r requirements.yml -p {{collections_destination|quote}}{{ ' -' + 'v' * ansible_verbosity if ansible_verbosity else '' }}
+      - name: fetch galaxy collections from collections/requirements.(yml/yaml)
+        command: ansible-galaxy collection install -r requirements{{ item.item.ext }} -p {{collections_destination|quote}}{{ ' -' + 'v' * ansible_verbosity if ansible_verbosity else '' }}
```

Comment:
this is very compact syntax, it took some thought for me to process it.

There are 2 `with_items` loops, and they're looping over a different thing. The syntax `item.ext` is in both loops because you have intentionally formatted the playbook `vars:` so that they will match.

The takeaway here is that your `when:` clause here will assure that we will only run `ansible-galaxy collection install` once for each file (`requirements.yml` or `requirements.yaml`). Since the user should only have one, we should only run the task once (which is good).